src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,73 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LOGIN_LENGTH = 6;
+    private static final int MIN_PASSWORD_LENGTH = 6;
+    private static final int MIN_USER_AGE = 18;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        validateUser(user);
+
+        return storageDao.add(user);
```

Comment:
Consider validating the user's data before adding them to the storage. The current implementation adds the user first without checking if the user data is valid. This could lead to invalid data being added to the storage. You can move the `validateUser(user)` call to the beginning of the `register` method.

Reply:
storageDao.add(user); comes after validateUser. If validation will fail it will not come to that line